benchmarks/threadsafe_let_block.rb

```diff
@@ -0,0 +1,304 @@
+require 'rspec/core'
+
+# switches between these implementations - https://github.com/rspec/rspec-core/pull/1858/files
+# benchmark requested in this PR         - https://github.com/rspec/rspec-core/pull/1858
+#
+# I ran these by adding "benchmark-ips" to ../Gemfile
+# then exported BUNDLE_GEMFILE to point t it
+# then ran `bundle exec rspec threadsafe_let_block.rb`
```

Comment:
For future referene, the `Gemfile` supports the git-ignored `Gemfile-custom` specifically so contributors (and core devs) can add one-off gems that aren't generally used by the travis build or by most contributors (such as benchmark-ips):

https://github.com/rspec/rspec-core/blob/9d12a3029f39250eb4affea076f82753d5ad5d55/Gemfile#L35

So there's no need to edit `Gemfile` or export `BUNDLE_GEMFILE` the next time if you ever contribute this kind of thing again :).